Sources/ArcGISToolkit/Components/Augmented Reality/Utilities/GeoTrackingSceneView.swift

```diff
@@ -0,0 +1,199 @@
+                            }
+                        case .localized:
+                            // Update the camera controller every time geo-tracking is localized,
+                            // to ensure the best experience.
+                            if !initialCameraIsSet, let currentLocation, let currentHeading {
+                                // Set the initial heading of scene view camera based on location
+                                // and heading. Geo-tracking requires 90 degrees rotation.
+                                updateCameraController(
+                                    location: currentLocation,
+                                    heading: currentHeading + 90,
+                                    altitude: currentLocation.position.z ?? 0
+                                )
+                                Task.detached { @MainActor in
+                                    initialCameraIsSet = true
+                                }
+                            }
+                        @unknown default:
+                            fatalError("Unknown ARGeoTrackingStatus.State")
+                        }
+                    }
+                
+                if initialCameraIsSet {
+                    sceneViewBuilder(sceneViewProxy)
+                        .cameraController(cameraController)
+                        .attributionBarHidden(true)
+                        .spaceEffect(.transparent)
+                        .atmosphereEffect(.off)
+                        .interactiveNavigationDisabled(true)
+                }
+            }
+            .ignoresSafeArea(.all)
+            .overlay {
+                ARCoachingOverlay(goal: .geoTracking)
+                    .sessionProvider(arViewProxy)
+                    .onCoachingOverlayActivate { _ in
+                        coachingOverlayIsActive = true
+                    }
+                    .onCoachingOverlayDeactivate { _ in
+                        coachingOverlayIsActive = false
+                    }
+                    .onCoachingOverlayRequestSessionReset { _ in
+                        // Reset the AR session to provide the best tracking performance.
+                        arViewProxy.session.run(configuration, options: .resetTracking)
+                    }
+                    .allowsHitTesting(false)
+            }
+        }
+        .observingInterfaceOrientation($interfaceOrientation)
+        .onAppear {
+            arViewProxy.session.run(configuration)
+        }
+        .onDisappear {
+            arViewProxy.session.pause()
+        }
+        .task {
+            for await location in locationDataSource.locations {
+                currentLocation = location
+            }
+        }
+        .task {
+            for await heading in locationDataSource.headings {
+                currentHeading = heading
+            }
+        }
+        .onReceive(calibrationViewModel.headingCorrections) { correction in
```

Comment:
If this logic is also duplicated in the WorldTracking version, then we should see if there is a way to break the logic out into a view modifier that can be reused. It might need to take a calibrationViewModel and a camera controller.

Reply:
I'll leave it unchanged for now. In my tests, once the geo-tracking is localized, it doesn't need adjustments and is pretty accurate, so maybe eventually we don't need this for geo-tracking.

The problem is I cannot find the right way to set the initial heading correctly. 😢 Will discuss in Slack tomorrow.

Reply:
issue 4992